test/dom/attr.test.js

```diff
@@ -1,71 +1,65 @@
 'use strict'
+const { DOMParser } = require('../../lib/dom-parser')
+const { DOMException } = require('../../lib/dom')
 
-var DOMParser = require('../../lib/dom-parser').DOMParser
-const assert = require('../assert')
-
-// Create a Test Suite
 describe('XML attrs', () => {
 	it('can properly set attribute', () => {
-		var root = new DOMParser().parseFromString('<xml/>', 'text/xml')
+		const root = new DOMParser().parseFromString('<xml/>', 'text/xml')
 			.documentElement
 		root.setAttribute('a', '1')
-		assert(root.attributes[0].localName, 'a')
+		expect(root.attributes[0].localName).toBe('a')
+
 		root.setAttribute('b', 2)
 		root.setAttribute('a', 1)
 		root.setAttribute('a', 1)
 		root.setAttribute('a', 1)
-		assert(root.attributes.length, 2)
-		try {
-			var c = root.ownerDocument.createElement('c')
+		expect(root.attributes.length).toBe(2)
+
+		const c = root.ownerDocument.createElement('c')
+		expect(() => {
 			c.setAttributeNode(root.attributes.item(0))
-		} catch (e) {
-			assert(e.code, 10)
-			return
-		}
-		assert.fail('expected error but none was thrown')
+		}).toThrow(new DOMException(10))
 	})
 
 	it('can properly set ns attribute', () => {
-		var root = new DOMParser().parseFromString(
+		const root = new DOMParser().parseFromString(
 			"<xml xmlns:a='a' xmlns:b='b' xmlns='e'><child/></xml>",
 			'text/xml'
 		).documentElement
-		var child = root.firstChild
+
+		const child = root.firstChild
 		child.setAttributeNS('a', 'a:a', '1')
 		child.setAttributeNS('b', 'b:b', '2')
 		child.setAttributeNS('b', 'b:a', '1')
-		assert(child.attributes.length, 3, 'after adding 3', child)
+		expect(child.attributes.length).toBe(3)
 		child.setAttribute('a', 1)
 		child.setAttributeNS('b', 'b:b', '2')
-		assert(child.attributes.length, 4, 'after adding 4 and one with namespace')
-		try {
-			var c = root.ownerDocument.createElement('c')
+		expect(child.attributes.length).toBe(4)
+
+		const c = root.ownerDocument.createElement('c')
+		expect(() => {
 			c.setAttributeNodeNS(root.attributes.item(0))
-		} catch (e) {
-			assert(e.code, 10, 'wrong error code')
-			return
-		}
-		assert.fail('expected error but none was thrown')
+		}).toThrow(new DOMException(10))
 	})
 
 	it('can properly override attribute', () => {
-		var root = new DOMParser().parseFromString(
+		const root = new DOMParser().parseFromString(
 			"<xml xmlns:a='a' xmlns:b='b' xmlns='e'><child/></xml>",
 			'text/xml'
 		).documentElement
 		root.setAttributeNS('a', 'a:a', '1')
-		assert(root.attributes.length, 4)
+		expect(root.attributes.length).toBe(4)
 		//not standart
 		//    	root.firstChild.setAttributeNode(root.attributes[0]);
-		//    	assert(root.attributes.length, 0);
+		//    	expect(root.attributes).toHaveLength(0);
```

Comment:
I think it would be really awesome if we could understand why these commented-out lines are here and if there is anything we can do to resolve them. These commented-out lines seem to have come from this commit back in 2012: c9f94a8c0c3bfd8eea5c2050bffed09d765f61f3

Reply:
My first guess: 
It's a test for something that is not specified in WC3 standard.
If it passes we should just activate and document it (in a different PR), since people might rely on it.
If it doesn't pass we should get rid of the comment :)

Will check it.

Reply:
Even though adding those lines leaves a passing test, it leaves the DOM in a broken state: the serialization of `rootNode` or `rootNode.firstChild` fails with `Attribute in use`.
Obviously it was not transferred to the new parent/owner correctly.

So I think we should just drop those lines. I don't think they hint to anything that is worth having a test for right now.

Reply:
I would kept the comment, ideally with as much historical context as we know, until we get a chance to test the functionality. As I said in <https://github.com/xmldom/xmldom/issues/72#issuecomment-761727690> this looks like untested functionality.

Reply:
> Even though adding those lines leaves a passing test, it leaves the DOM in a broken state

@karfau I missed that comment, guess you are right to remove it then. Hope we get a chance to test it someday:)